lib/mox.ex

```diff
@@ -183,6 +183,15 @@ defmodule Mox do
   `$callers` is used to determine the process that actually defined the
   expectations.
 
+  #### Explicit allowances as lazy/deferred functions
+
+  Under some circumstances, the process might not have been already started
+  when the allowance happens. In such a case, you might specify the allowance
+  as a lazy/deferred function call in a form `(-> pid())`. This function
+  would be resolved late, at the very moment of dispatch. Please note, that no
+  additional diagnostics could have been provided for lazy calls, and it would
+  either succeed, or fail with `Mox.UnexpectedCallError`. 
```

Comment:
```suggestion
  #### Explicit allowances as lazy/deferred functions

  Under some circumstances, the process might not have been already started
  when the allowance happens. In such a case, you might specify the allowance
  as a function in the form `(-> pid())`. This function would be resolved late,
  at the very moment of dispatch. If the function does not return an existing
  PID, it will fail `Mox.UnexpectedCallError`. 
```